Search recipes

LaviniaBinisor opened the pull request:
Add the functionality for the SearchText in a VStack .

Commit:
Create the functionality for SearchText.
Move the functionality in a VStack.

LaviniaBinisor committed:
Update README.md

LaviniaBinisor committed:
Update README.md

BinisorIonut commented on MyRecipes/ViewModels/RecipesViewModel.swift:
```diff
@@ -9,6 +9,17 @@ import Foundation
+        guard !searchText.isEmpty else { return recipes }
+        
+        return recipes.filter { $0.name.localizedCaseInsensitiveContains(searchText) }
+//        return recipes.filter { recipe in
```
is there a reason for keeping the commented code here?

LaviniaBinisor replied:
done

BinisorIonut commented on MyRecipes/ViewModels/RecipesViewModel.swift:
```diff
@@ -26,5 +37,8 @@ extension RecipesViewModel {
             recipes[index] = recipe
         }
     }
+//        
```
same here, do you still need this commented code?

LaviniaBinisor replied:
done

BinisorIonut commented on MyRecipes/Views/Home/HomeView.swift:
```diff
@@ -13,10 +13,19 @@ struct HomeView: View {
             }
-            .navigationTitle("My Recipes")
+            .searchable(text: $recipesVM.searchText, prompt: "Find a recipe")
+            .navigationViewStyle(.stack)
```
Since your is iOS 16 or higher, you can use NavigationStack instead of NavigationView, and that means you no longer need  .navigationViewStyle(.stack) modifier

LaviniaBinisor replied:
done

BinisorIonut replied:
you still have this comment here

Commit:
Refactor code.
Instead of the search func, I created a computed property that does the filtering.

Commit:
I changed the deployment target of the application, from 16.0 to 17.0.

Commit:
the navigationViewStack is not longer necessary, because NavigationView also contains this functionality.

Commit:
Merge branch 'main' into SearchRecipes

# Conflicts:
#	README.md
